tests/org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/macro/AbstractReusableActiveAnnotationTests.xtend

```diff
@@ -3363,6 +3363,87 @@ abstract class AbstractReusableActiveAnnotationTests {
 		]
 	}
 	
+	@Test def void testMoveExpressionToTypeParameterDeclarator() {
+		assertProcessing(
+			'myannotation/AbstractAnnotation.xtend' -> '''
+				package myannotation
+				
+				import java.lang.annotation.ElementType
+				import java.lang.annotation.Target
+				import org.eclipse.xtend.lib.macro.AbstractClassProcessor
+				import org.eclipse.xtend.lib.macro.Active
+				import org.eclipse.xtend.lib.macro.TransformationContext
+				import org.eclipse.xtend.lib.macro.declaration.MethodDeclaration
+				import org.eclipse.xtend.lib.macro.declaration.MutableClassDeclaration
+				
+				@Target(ElementType.TYPE)
+				@Active(SomeAnnotationProcessor)
+				annotation SomeAnnotation {
+				}
+				
+				class SomeAnnotationProcessor extends AbstractClassProcessor {
+				
+					static def copyMethod(MutableClassDeclaration clazz, MethodDeclaration method, extension TransformationContext context) {
+				
+						val methodDeclaration = clazz.addMethod(method.simpleName) [
+							it.abstract = method.abstract
+							it.docComment = method.docComment
+							it.visibility = method.visibility
+							it.synchronized = method.synchronized
+							it.deprecated = method.deprecated
+							it.static = method.static
+							it.exceptions = method.exceptions
+							it.varArgs = method.varArgs
+							it.strictFloatingPoint = method.strictFloatingPoint
+						]
+				
+						for (t : method.typeParameters) {
+							val result = methodDeclaration.addTypeParameter(t.simpleName, t.upperBounds)
+							result.primarySourceElement = t
+						}
+				
+						methodDeclaration.returnType = method.returnType
+				
+						for (p : method.parameters) {
+							methodDeclaration.addParameter(p.simpleName, p.type)
+						}
+				
+						return methodDeclaration
+				
+					}
+				
+					override doTransform(MutableClassDeclaration annotatedClass, extension TransformationContext context) {
+						
+						for (m : annotatedClass.declaredMethods) {
+							
+							val newMethod = annotatedClass.copyMethod(m, context)
+							newMethod.simpleName = newMethod.simpleName + "_copiedMethod"
+							newMethod.body = m.body
+							m.body = ''��'// new method''��'
+							
+						}
+						
+					}
+				
+				}
+			''',
+			'myusercode/UserCode.xtend' -> '''
+				package myusercode
```

Comment:
You would need to not simply copy the parameter's TypeReference but let it point to the copied TypeParameter.